contributingGuides/TYPESCRIPT_STYLE.md

```diff
@@ -0,0 +1,857 @@
+  weatherApiTemperature: number;
+  weatherApiCity: string;
+}
+
+// Good
+type Weather = {
+  temperature: number;
+  city: string;
+}
+```
+
+## Types and Interfaces
+
+Use **type** by default.
+
+```ts
+// Bad
+interface Report {
+    // properties
+}
+
+// Good
+type Report = {
+    // properties
+}
+```
+
+TypeScript supports both type and interface expressions. Types can be used more broadly: to name primitives, unions, objects and any other type. As these two forms are practically equivalent, it's best to select one over the other to maintain consistency and reduce variation.
+
+One drawback of using interfaces is that they can be declared multiple times and are merged. This can lead to unintended behavior.
+
+```ts
+// Merging interfaces
+interface Foo {
+  a: number;
+}
+interface Foo {
+  b: number;
+}
+
+// Foo is now { a: number; b: number; }
+```
+
+Use interfaces only when you encounter a situation where `type` does not fulfill your requirements. Example scenario is when you need to modify or extend an interface from a third-party library that lacks typings or has incorrect typings. 
+
+When extending or correcting interfaces from third-party libraries, always use [module augmentation](https://www.typescriptlang.org/docs/handbook/declaration-merging.html#module-augmentation) to maintain code consistency and avoid potential issues.
+
+```ts
+// Third-party library with missing or incorrect typings
+interface LibraryComponentProps {
+  // Existing incorrect or incomplete typings
+}
+
+// Your code
+declare module 'library/path' {
+  interface LibraryComponentProps {
+    // Add or modify the typings
+    additionalProp: string;
+  }
+}
+```
+
+## Enums and Union Types
+
+Use **union types** instead of enums. Opt for union types whenever possible. Union types provide flexibility and can handle varying data types and dynamic value sets.
+
+```ts
+// Bad
+enum Colors {
+  Red,
+  Green,
+  Blue
+}
+
+// Good
+type Colors = 'red' | 'green' | 'blue';
+```
+You can also define a union type by deriving the type from an array. It's a great choice when seeking an easily iterable and directly usable array in code while keeping one source of truth.
+
+```ts
+// Bad
+const COLORS = ['red', 'green', 'blue'] as const;
+type Colors = 'red' | 'green' | 'blue';
+
+// Good
+import { TupleToUnion } from 'type-fest';
+
+const COLORS = ['red', 'green', 'blue'] as const;
+type Colors = TupleToUnion<typeof COLORS>;
+```
+
+Defining union types using an object offers several benefits. This approach it's a great choice when **enum-like identifiers are needed**, while maintaining one source of truth and improving readability.
+
+```ts
+// Bad
+const COLORS = {
+    Red: 'red',
+    Green: 'green',
+    Blue: 'blue',
+} as const;
+type Colors = 'red' | 'green' | 'blue';
+
+// Good
+import { ValueOf } from 'type-fest';
+
+const COLORS = {
+    Red: 'red',
+    Green: 'green',
+    Blue: 'blue',
+} as const;
+type Colors = ValueOf<typeof COLORS]>
```

Comment:
That square bracket looks lonely or like it shouldn't be there 🤔 `<typeof COLORS]>`